src/gsy_e/models/market/two_sided.py

```diff
@@ -318,7 +318,7 @@ def match_recommendations(
                 #  on a matching algorithm setting
                 if is_external_matching_enabled():
                     # re-raise exception to be handled by the external matcher
-                    raise InvalidBidOfferPairException from invalid_bop_exception
+                    raise invalid_bop_exception
```

Comment:
Here we're conditionally raising a new exception, this is swallowing the exception message and not forwarding it to the higher level code, so I reraised the same exception